mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/navansatt/ktor/AuthorizedRoute.kt

```diff
@@ -6,38 +6,68 @@ import no.nav.mulighetsrommet.api.navansatt.model.Rolle
 
 fun Route.authorize(
     requiredRole: Rolle,
-    vararg additionalRoles: Rolle,
     build: Route.() -> Unit,
 ): Route {
-    val routeRoles = listOf(requiredRole, *additionalRoles)
+    val routeRoles = listOf(AnyRoles(setOf(requiredRole)))
+    val authorizedRoute = createChild(NavAnsattAuthorizationRouteSelector(RequiredRoles(routeRoles)))
 
-    val authorizedRoute = createChild(NavAnsattAuthorizationRouteSelector(routeRoles))
+    val parentRoles = generateSequence(authorizedRoute) { it.parent }
+        .mapNotNull { it.attributes.getOrNull(AuthorizedRolesKey) }
+        .map { it.roles }
+        .flatten()
+        .toList()
+        .reversed()
+        .distinct()
+
+    val combinedRoles = RequiredRoles(parentRoles + routeRoles)
+
+    authorizedRoute.attributes.put(AuthorizedRolesKey, combinedRoles)
+
+    authorizedRoute.install(NavAnsattAuthorizationPlugin) {
+        requiredRoles = combinedRoles
+    }
+
+    authorizedRoute.build()
+    return authorizedRoute
+}
+
+fun Route.authorize(
+    anyOf: Set<Rolle> = emptySet(),
+    allOf: Set<Rolle> = emptySet(),
+    build: Route.() -> Unit,
+): Route {
```

Comment:
Kan se ut som at vi kan slå sammen disse to `authorize`-funksjonene ved å endre signaturen litt:

```
fun Route.authorize(
    requiredRole: Rolle? = null,
    anyOf: Set<Rolle> = emptySet(),
    allOf: Set<Rolle> = emptySet(),
    build: Route.() -> Unit,
): Route {
```

Da må gjøre `requiredRole` påkrevd når den ikke er null under `buildList`en under, men ellers kan det se ut som endringen er bakoverkompatibel :crossed_fingers: